doc/man3/SSL_alloc_buffers.pod

```diff
@@ -0,0 +1,62 @@
+=pod
+
+=head1 NAME
+
+SSL_free_buffers, SSL_alloc_buffers - manage SSL structure buffers
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ int SSL_free_buffers(SSL *ssl);
+ int SSL_alloc_buffers(SSL *ssl);
+
+=head1 DESCRIPTION
+
+SSL_free_buffers() frees the read and write buffers of the given B<ssl>.
+SSL_alloc_buffers() allocates the read and write buffers of the given B<ssl>.
+Using these functions can save around 34k per SSL connection.
+
+The SSL_MODE_RELEASE_BUFFERS mode releases read or write buffers when the
+buffers have been drained. If this is too aggressive, the buffers can be
+freed and allocated using these functions.
+
+After freeing the buffers, the buffers are automatically reallocted upon a
+new read or write. The SSL_alloc_buffers() does not need to be called, but
+can be used to control when the buffers are allocated.
+
+=head1 RETURN VALUES
+
+The following return values can occur:
+
+=over 4
+
+=item Z<>0 (Failure)
+
+The SSL_free_buffers() function returns 0 when there is pending data to be
+read or written. The SSL_alloc_buffers() function returns 0 when there is
+an allocation failure.
+
+=item Z<>1 (Success)
+
+The SSL_free_buffers() function returns 1 when the buffers are or were
+freed. The SSL_alloc_buffers() function returns 1 if the buffers are or
+were allocated.
```

Comment:
Either the buffers are already allocated when the call was issued, or they were allocated by the call. I can make this a bit more clear.